trade-finance-manager-api/src/v1/validation/validateIds.js

```diff
@@ -0,0 +1,53 @@
+const validator = require('validator');
+const teams = require('../../constants/teams');
+
+const isValidMongoId = (mongoId) => (mongoId ? validator.isMongoId(mongoId) : false);
+
+const isValidUkefNumericId = (ukefId) => {
+  const id = parseInt(ukefId, 10);
+
+  if (Number.isNaN(id)) {
+    return false;
+  }
+
+  const regex = /^\d{10}$/;
+
+  return regex.test(ukefId);
+};
+
+const isValidPartyUrn = (partyUrn) => {
+  const regex = /^\d{8}$/;
+
+  return regex.test(partyUrn);
+};
+
+const isValidNumericId = (numericId) => {
+  const id = parseInt(numericId, 10);
+
+  if (Number.isNaN(id)) {
+    return false;
+  }
+
+  const regex = /^\d+$/;
+
+  return regex.test(numericId);
+};
+
+const isValidCurrencyCode = (currencyCode) => (currencyCode ? validator.isISO4217(currencyCode) : false);
+
+const sanitizeUsername = (username) => validator.escape(username);
```

Comment:
We have one endpoint that uses username in a URL. This comes from a graphql `mutation-update-lead-underwriter` which could come from user input. However username's have no validation when they are saved and can be made up of any characters so it is hard to validate and narrow down what an allowed username is so right now all I do is escape the username. Would welcome any thoughts here

Reply:
Can we ensure following symbols are escapes please (ideally unit test to satisfy these scenarios):

* $
* ../
* \
* '
* <
* >

Reply:
This function escapes `\` `/` `<` `"` `'` which I think should cover all attacks in SSRF apart from trying to insert localhost, although none of our endpoints using this username in an endpoint are going to be vulnerable to a `localhost` or local IP attack

Reply:
(In my next commit will have a test for all these characters)